src/Std/Net/Addr.lean

```diff
@@ -193,5 +194,22 @@ def port : SocketAddress → UInt16
 
 end SocketAddress
 
+/--
+Represents an interface address, including details such as the interface name,
+whether it is internal, the associated address, and the network mask.
+-/
+structure InterfaceAddress where
+  name : String
+  isInternal : Bool
+  address : SocketAddress
+  netMask : SocketAddress
+  deriving Inhabited, DecidableEq
+
+/--
+Retrieves a list of interface addresses.
```

Comment:
1. It retrieves an `Array` but that's not so important
2. We should make clear which interface addresses it retrieves, is it all of them, what about disabled ones, what happens if an interface has multiple addresses? I hope the answers to these questions to be yes, they are also shown and we get multiple entries.